File: gfg-solutions/armstrong-numbers.py
# Question Link : https://www.geeksforgeeks.org/problems/armstrong-numbers2727/1

# Time Complexity : 
# Space Complexity : 

# Iterative Solution

class Solution:
    def evaluate(self, num):
        res = 0
        while(num!=0):  
            currentDigit = num%10
            num = num//10
            res = res+ currentDigit**3
        
        return res
        
    def armstrongNumber (self, n):
        result = self.evaluate(n)
        return "Yes" if result == n else "No"

File: gfg-solutions/armstrong-numbers.cpp
// Question Link : https://www.geeksforgeeks.org/problems/armstrong-numbers2727/1

// NOTE: 371 is an Armstrong number since 33 + 73 + 13 = 371

/*
 Time Complexity : 
 Space Complexity : 
*/

// Recursive Solution

class Solution {
  public:
    int evaluate(int n) {
        if (n == 0) {
            return 0;
        }
        if (n > 0) {
            return (pow(n%10, 3) + evaluate(floor(n/10)));
        }
    }

    string armstrongNumber(int n){
        return n == evaluate(n) ? "Yes" : "No";
    }
};

// Iterative Solution
class Solution {
  public:
    int evaluate(int num) {
        int result = 0;
        while(num != 0) {
            int currentDigit = num % 10;
            num = num/10;
            result += pow(currentDigit , 3);
        }
        return result;
    }
    
    string armstrongNumber(int n){
        int ans = evaluate(n);
        
        return ans == n ? "Yes" : "No";
    }
};